### ISSUE TITLE
Lightgbm  UserWarning: `step` X is already reported.

### ISSUE THREAD

**Issue by rm17tink** [optuna/optuna#3680]
### Expected behavior

I am getting warnings for every step using lightgbm and optuna on python 3.8 

### Environment

- Optuna version: '2.10.1'
- lightgbm: '3.3.2'
- Python version: 3.8
- OS: Linux 
- (Optional) Other libraries and their versions:

### Error messages, stack traces, or logs

```shell
_warnings.warn(
/opt/conda/envs/python38/lib/python3.8/site-packages/optuna/trial/_trial.py:590: UserWarning: The reported value is ignored because this `step` 22 is already reported._

X 1000
```

### Steps to reproduce

```

import optuna  
from optuna.integration import LightGBMPruningCallback 
from optuna.samplers import TPESampler
from lightgbm import early_stopping

def objective(trial, cv_n_splits, X, y, X_test, y_test):
    global model
    param_grid = {
         #         "device_type": trial.suggest_categorical("device_type", ['gpu']),
        "n_estimators": trial.suggest_int("n_estimators", 100, 7000, step=100),
        #"n_estimators": trial.suggest_categorical("n_estimators", [10000]),
        "verbose": -1,  # mutes the [WARNING] output
        "boosting_type": "gbdt", #dart
         "seed": 42,
        #"scale_pos_weight": 583, #(makes performance terrible)
        #"is_unbalance": "true",  #added ( makes performance terrible)
        "learning_rate": trial.suggest_float("learning_rate", 0.1, 0.3),
        "num_leaves": trial.suggest_int("num_leaves", 20, 3000, step=20),
        "max_depth": trial.suggest_int("max_depth", 3, 12),
        #min_data_in_leaf
        "min_child_samples": trial.suggest_int("min_child_samples", 200, 10000, step=100), 
        "max_bin": trial.suggest_int("max_bin", 200, 300),
        #lambda_l1
        "reg_alpha": trial.suggest_int("reg_alpha", 0, 100, step=5),
        #lambda_l2
        "reg_lambda": trial.suggest_int("reg_lambda", 0, 100, step=5),
        #min_gain_to_split
        "min_split_gain": trial.suggest_float("min_split_gain", 0, 15),
        # bagging_fraction
        "subsample": trial.suggest_float("subsample", 0.2, 0.90, step=0.1), 
        # bagging_freq
        "subsample_freq": trial.suggest_categorical("subsample_freq", [1]),
        #feature_fraction
        "colsample_bytree": trial.suggest_float("colsample_bytree", 0.2, 0.90, step=0.1),      
    }
    
    
    cv = StratifiedKFold(n_splits=cv_n_splits, shuffle=True, random_state=1121218)
    
    #CV storage value
    auc_scores_train   = np.empty(5)
    auc_scores_valid   = np.empty(5)
    auc_scores_test    = np.empty(5)
    valid_train_spread = np.empty(5)
    valid_test_spread  = np.empty(5)
    
    
    for idx, (train_idx, valid_idx) in enumerate(cv.split(X, y)):
        X_train, X_valid = X.iloc[train_idx], X.iloc[valid_idx]
        y_train, y_valid = y[train_idx], y[valid_idx]

        model = lgbm.LGBMClassifier(objective="binary", **param_grid)
        model.fit(
            X_train,
            y_train,
            eval_set=[(X_valid, y_valid)],
            eval_metric="auc", 
            # Add a pruning callback
            callbacks=[early_stopping(100),LightGBMPruningCallback(trial, "auc")], 
        )
    
        #TRAINING INFO 
        train_preds = model.predict_proba(X_train)
        auc_scores_train[idx] = roc_auc_score(y_train, train_preds[:,1])
        trial.set_user_attr("auc_scores_train", np.mean(auc_scores_train))    
        
        
        #TEST INFO (measurement we maximize in the trials)
        X_test = X_test
        y_test = y_test
        test_preds = model.predict_proba(X_test)
        auc_scores_test[idx] = roc_auc_score(y_test, test_preds[:,1])
        trial.set_user_attr("auc_scores_test", np.mean(auc_scores_test))    

        
        #VALID INFO (measurement we maximize in the trials)
        valid_preds = model.predict_proba(X_valid)
        auc_scores_valid[idx] = roc_auc_score(y_valid, valid_preds[:,1])
    
        #Valid Train Spread 
        valid_train_spread[idx] = (auc_scores_train[idx] - auc_scores_valid[idx])
        trial.set_user_attr("valid_train_spread", np.mean(valid_train_spread))
        
        #Test  Valid  Spread 
        valid_test_spread[idx] = (auc_scores_test[idx] - auc_scores_valid[idx])
        trial.set_user_attr("valid_test_spread", np.mean(valid_test_spread))
        
    return np.mean(auc_scores_valid)
```

```


_if __name__ == "__main__":
    study = optuna.create_study(pruner=optuna.pruners.MedianPruner(n_warmup_steps=10)
                                , direction="maximize", study_name="LGBM Classifier"
                                ,sampler=TPESampler(seed=42))
    #supress logging warnings
    #optuna.logging.set_verbosity(optuna.logging.WARNING)
    func = lambda trial: objective(trial 
                                    #CV_n_splits
                                         ,5
                                         # pass in the dataset for CV
                                         ,train_valid_downsampled[feature_columns] 
                                         ,train_valid_downsampled[target]
                                         #test set 
                                         ,test_downsampled[feature_columns] 
                                         ,test_downsampled[target]
                                        )
    study.optimize(func, n_trials=2, callbacks=[callback])

```

**Comment by rm17tink** [comment]
Could you tell me how to repress it then?

**Comment by nzw0301** [comment]
I think either way

1. Not report intermediate values multiple times https://github.com/optuna/optuna/issues/3680#issuecomment-1158666391
2. Change the logger level as commented out in the code.

**Comment by Kaschi14** [comment]
Could this still be a bug?
My Training / Val Loop consists of epochs and iterates of the dataset batchwise (and Im using tqdm):
`for epoch in range(epochs):
       with tqdm(self.val_dataloader, unit="batch") as tepoch:
                with torch.no_grad():
                    for i, data in enumerate(tepoch):
                           ...
                           loss_temp = self.criterion(outputs, labels).item()
                           ...
                           # For Optuna Pruning
                           self.trial.report(loss_temp, i)
                           # Handle pruning based on the intermediate value.
                           if self.trial.should_prune():
                               raise optuna.TrialPruned()
`
So I report the loss of the current batch.
But as soon as I'm in the validation of the 2nd epoch I get the warning and it in fact doesn't look at the loss anymore, even though it should.
Should I give the step value a unique value? maybe something like this:
self.trial.report(loss_temp, i + epoch * len(tepoch))

**Comment by nzw0301** [comment]
Please use https://github.com/optuna/optuna/discussions to ask a question that is not related to this issue.
